<a href="https://colab.research.google.com/github/shrutiagarwal2615/Shopify_DS_Intern_Challenge/blob/main/Shopify.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

#Question 1: 
Given some sample data, write a program to answer the following: click here to access the required data set

On Shopify, we have exactly 100 sneaker shops, and each of these shops sells only one model of shoe. We want to do some analysis of the average order value (AOV). When we look at orders data over a 30 day window, we naively calculate an AOV of $3145.13. Given that we know these shops are selling sneakers, a relatively affordable item, something seems wrong with our analysis. 


In [ ]:

# there are two diffrent types of orders looking at the data, a lot of values for item quantity are less than 10 and some are greater than 1000 therefore the average does not reflect real orders 
# because the sample is not coherent. the solution would be to divide the sample and calculate two diffrent averages: Business-to-Customer and Business-to-Business

#define main
def main():

	B2Bamount =0.0
	B2Camount = 0.0
	B2Borders =0.0
	B2Corders = 0.0
	Totalamount =0.0
	
	with open("/content/Shopify_Intern_Data.csv", "r") as file:
		next(file)
		for line in file:
			order_id, shop_id, user_id, order_amount, total_items, payment_method, created_at = line.split(",")
			order_amount = int(order_amount)
			total_items = int(total_items)
			Totalamount += order_amount
			if (total_items > 1000): #diving orders into two categories (B2B and B2C)
				B2Bamount += order_amount
				B2Borders += 1
			else:
				B2Camount += order_amount
				B2Corders += 1

	print("AOV of B2B is " , B2Bamount / B2Borders)
	print("AOV of B2C is " , B2Camount / B2Corders)
	print("AOV of total orders is",Totalamount/(B2Borders+B2Corders) )
#define run
if __name__ == "__main__":
	main()

AOV of B2B is  704000.0
AOV of B2C is  754.0919125025085
AOV of total orders is 3145.128


##**A**. Think about what could be going wrong with our calculation. Think about a better way to evaluate this data. 

### As we can see that there were certain high values of order amounts in the data as compared to other values which are usually below 10, so the AOV value is probably coming from the mean of all the orders and ignoring the large deviation in order amounts. 
###To better evaluate this data, after validating that data records the minimum value of order amount is 90 and maximum is 704000 and the frequency of the orders with  high order amount is having a fixed pattern like same order items from one or two stores ordered daily. So, i decided to divide the orders into two categories B2B and B2C on the basis that item ordered >1000 then its B2B.

##**B**. What metric would you report for this dataset?


###AOV is the average order value calculated by dividing the total revenue with the total number of orders. Since, we identified in the data that there are two major categories of order placed, B2B and B2C, so we need to calculate AOV for both categories and these numbers are now more reasonable.

##**C**.What is its value?

AOV of B2B is  704000.0

AOV of B2C is  754.0919125025085  
This AOV value now seems more aligned and reasonable with the order amounts for business to customer transactions. 
